common/compiler/solidity.go

```diff
@@ -146,6 +147,10 @@ func runsolc(cmd *exec.Cmd, source string) (map[string]*Contract, error) {
 		if err := json.Unmarshal([]byte(info.Devdoc), &devdoc); err != nil {
 			return nil, fmt.Errorf("solc: error reading dev doc: %v", err)
 		}
+		var metadata interface{}
+		if err := json.Unmarshal([]byte(info.Metadata), &metadata); err != nil {
```

Comment:
It would be backwards-compatible if you put `if info.Metadata != "" { ... }` around the json.Unmarshal call. Please add this.